### DISCUSSION TITLE
简单做了个 Naive UI px 转 rem 的小工具 Make a tool for Naive UI `px` to convert `rem`

### DISCUSSION THREAD

**Discussion by A81N9** [tusen-ai/naive-ui#6664] [General]
### Clear and concise description of the problem

简单做了个 Naive UI `px` 转 `rem` 的小工具
Make a tool for Naive UI `px` to convert `rem`

[Naive UI Rem Converter](//naive-px2rem.vercel.app)

### Suggested solution

1. 输入基准值并点击 `Convert` 按钮
Enter the benchmark value and click the `Convert` button.

3. 可在右侧切换组件名称及主题
You can switch the component name and theme on the right side.

5. 复制配置到 `NConfigProvider` 的 `theme-overrides` 属性
Copy the configuration to the `theme-overrides` attribute of `NConfigProvider`.

![image](https://github.com/user-attachments/assets/8483c02e-8110-455f-a79c-50279a881daf)

### Additional context

#4250 #6096 #4738 #3940 #3841 #3288 #3108 #2102 #802 

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/tusen-ai/naive-ui/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://www.naiveui.com/en-US).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by A81N9** [comment]
@07akioni 感觉可以在 `NThemeEditor` 中集成此功能

**Comment by A81N9** [comment]
```
Space: {
  gapSmall: '0.25rem 0.5rem',
  gapMedium: '0.5rem 0.75rem',
  gapLarge: '0.75rem 1rem',
},
```

发现 NSpace 设置 rem 不起作用

**Comment by A81N9** [comment]
```
"Flex": {
    "gapSmall": "0.25rem 0.5rem",
    "gapMedium": "0.5rem 0.75rem",
    "gapLarge": "0.75rem 1rem"
  }
```
NFlex 设置 rem 也不生效

**好像涉及到 gap 属性的值设置 rem 都不生效**

**Comment by jc840031139** [comment]
设置之后 select面板不会展示

**Comment by jahnli** [comment]
Thanks to you, I moved it to the discussion area for more people to see.

**Comment by Rolaka** [comment]
Some uses of VirtualList cannot be converted, example:
https://github.com/tusen-ai/naive-ui/blob/942de11cd755288b2f8fc55a048ad76a5b152d46/src/_internal/select-menu/src/SelectMenu.tsx#L518-L529

The itemSize and padding here are solved from the styles using seemly's depx, by：`Number(value.slice(0, value.length - 2));`

So the Select component can override the `InternalSelection` 和`InternalSelectMenu`，Forced to specify px as the fix, example：
https://github.com/tusen-ai/naive-ui/blob/942de11cd755288b2f8fc55a048ad76a5b152d46/themes/tusimple/src/theme-overrides-light.ts#L246-L258

A dirtier option might be to traverse the theme for convert after getting the screen size at init time.

**Comment by A81N9** [comment]
Simple implementation of conversion functions.

```vue
<script setup>
import { darkTheme } from 'naive-ui'

function toRem(theme, options = { rootValue: 16, unitPrecision: 6 }) {
  const { rootValue, unitPrecision } = options

  function convert(value) {
    if (typeof value === 'object' && value !== null) {
      return Object.entries(value).reduce(
        (acc, [key, val]) => {
          acc[key] = convert(val)
          return acc
        },
        Array.isArray(value) ? [] : {}
      )
    }

    if (typeof value === 'string' && value.includes('px')) {
      return value.replace(/(\d+(\.\d+)?)px/g, (_, p1) => `${Number((p1 / rootValue).toFixed(unitPrecision))}rem`)
    }

    return value
  }

  return convert(theme)
}
</script>

<template>
  <n-config-provider :theme="toRem(darkTheme)">
    
  </n-config-provider>
</template>
```

**Comment by Hannicke** [comment]
<img width="711" height="101" alt="image" src="https://github.com/user-attachments/assets/6c1f6581-976d-4eb5-8e7b-4436fcc861bb" />

**Reply by Hannicke** [reply]
项目地址的jquery挂了

**Reply by A81N9** [reply]
已更新
